Mark CTE supported (#1629)
vitessio/website: content/en/docs/19.0/reference/compatibility/mysql-compatibility.md

A similar effect can be achieved by using a database name like `mykeyspace:-80@rdonly` in your MySQL application client connection string.

### Common Table Expressions

> Vitess does not yet support recursive CTEs. 
> Non-recursive CTEs are supported, with the limitation that CTE aliases cannot be the same as a base table used in the CTE. 

Reviewer: ```suggestion
Vitess supports Non-recursive CTEs with the limitation that CTE aliases cannot be the same as a base table used in the CTE. 
Currently, Vitess does not have support for recursive CTEs. 
```
